components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -52,6 +53,22 @@ pub fn week_day(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<In
     Ok(Some(i64::from(day)))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn day_of_year(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Int>> {
+    if t.is_none() {
+        return Ok(None);
+    }
+    let t = t.as_ref().unwrap();
+    if t.invalid_zero() {
+        return ctx
+            .handle_invalid_time_error(Error::incorrect_datetime_value(&format!("{}", t)))
```

Comment:
`Error::incorrect_datetime_value(t)` is enough.

Reply:
updated